persistence/pre_persistence/account.go

```diff
@@ -0,0 +1,377 @@
+package pre_persistence
+
+import (
+	"bytes"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/types"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"math/big"
+)
+
+const (
+	ServiceNodeStakePoolName = "SERVICE_NODE_STAKE_POOL"
+	AppStakePoolName         = "APP_STAKE_POOL"
+	ValidatorStakePoolName   = "VALIDATOR_STAKE_POOL"
+	FishermanStakePoolName   = "FISHERMAN_STAKE_POOL"
+	DAOPoolName              = "DAO_POOL"
+	FeePoolName              = "FEE_POOL"
+)
+
+func (m *PrePersistenceContext) AddPoolAmount(name string, amount string) error {
+	cdc := Cdc()
+	p := Pool{}
+	db := m.Store()
+	key := append(PoolPrefixKey, []byte(name)...)
+	val, err := db.Get(key)
+	if err != nil {
+		return err
+	}
+	err = cdc.Unmarshal(val, &p)
+	if err != nil {
+		return err
+	}
+	s, err := StringToBigInt(p.Account.Amount)
+	if err != nil {
+		return err
+	}
+	s2, err := StringToBigInt(amount)
+	if err != nil {
+		return err
+	}
+	s.Add(s, s2)
+	p.Account.Amount = BigIntToString(s)
+	bz, err := cdc.Marshal(&p)
+	if err != nil {
+		return err
+	}
+	return db.Put(key, bz)
+}
+
+func (m *PrePersistenceContext) SubtractPoolAmount(name string, amount string) error {
```

Comment:
The only difference between this and the function above is one line so you could create a helper and remove copy-pasted code. Tested here: https://go.dev/play/p/7pB70u5dWXS
Example:

```
func (m *PrePersistenceContext) AddPoolAmount(name string, amount string) error {
	add := func(s *big.Int, s1 *big.Int) {
		s.Add(s, s1)
	}
	m.operationPoolAmount(name, amount, add)
}

func (m *PrePersistenceContext) operationPoolAmount(name string, amount string, op func(*big.Int, *big.Int)) error {
	cdc := Cdc()
	p := Pool{}
	db := m.Store()
	key := append(PoolPrefixKey, []byte(name)...)
	val, err := db.Get(key)
	if err != nil {
		return err
	}
	err = cdc.Unmarshal(val, &p)
	if err != nil {
		return err
	}
	s, err := StringToBigInt(p.Account.Amount)
	if err != nil {
		return err
	}
	s2, err := StringToBigInt(amount)
	if err != nil {
		return err
	}
	op(s, s2)
	p.Account.Amount = BigIntToString(s)
	bz, err := cdc.Marshal(&p)
	if err != nil {
		return err
	}
	return db.Put(key, bz)
}
```